docs/concepts/models/sql_models.md

```diff
@@ -40,20 +43,70 @@ SELECT
 FROM raw.restaurants AS r
 JOIN countries AS c
   ON r.id = c.restaurant_id;
+
+/*
+  Optional post-statements that will run after the model's query.
+  You should NOT do things that cause side effects that could error out when
+  executed concurrently with other statements, such as creating physical tables.
+*/
+UNCACHE TABLE countries;
 ```
 
-### Model DDL
-The `MODEL` DDL is used to specify metadata about the model such as name, [kind](./model_kinds.md), owner, cron, and others. The Model statement should be the first statement in your model SQL file.
+### `MODEL` DDL
+The `MODEL` DDL is used to specify metadata about the model such as its name, [kind](./model_kinds.md), owner, cron, and others. This should be the first statement in your SQL-based model's file.
+
+Refer to `MODEL` [properties](./overview.md#properties) for the full list of allowed properties.
+
+### Optional pre/post-statements
+Optional pre/post-statements can help prepare the model's query and execute "clean-up" actions after it has successfully executed, respectively. For example, you might create temporary views, set permissions, or evict previously cached tables from memory.
+
+However, be careful not to run any statement that could conflict with the execution of another statement when ran concurrently, such as creating a physical table.
+
+### The model query
+The model must contain a standalone query, which can be a single `SELECT` expression, or multiple `SELECT` expressions combined with the `UNION`, `INTERSECT`, or `EXCEPT` operators. The result of this query will be used to populate the model's table or view.
+
+## Python-based definition
 
-Refer to `model` [properties](./overview.md#properties) for the full list of allowed properties.
+The Python-based definition of SQL models consists of a single python function, decorated with SQLMesh's `@model` [decorator](https://wiki.python.org/moin/PythonDecorators). The decorator is required to have the `is_sql` keyword argument set to `True` to distinguish it from [Python models](./python_models.md) that return DataFrame instances.
 
-### Optional statements
-Optional SQL statements can help you prepare the model's `SELECT` statement. For example, you might create temporary tables or set permissions.
+This function's return value serves as the model's query, and it must be either a SQL string, or a [SQLGlot expression](https://github.com/tobymao/sqlglot/blob/main/sqlglot/expressions.py). The `@model` decorator is used to define the model's [metadata](#MODEL-DDL) and, optionally, its pre/post-statements that are also in the form of SQL strings or SQLGlot expressions.
```

Comment:
Remove comma from "either a SQL string, or a".

"...optionally, pre/post-statements in the form of SQL strings..."